pkg/controller/registry/resolver/resolver.go

```diff
@@ -554,27 +554,32 @@ func (r *SatResolver) sortChannel(bundles []*Operator) ([]*Operator, error) {
 		}
 	}
 
-	if len(headCandidates) != 1 {
-		// TODO: more context in error
-		return nil, fmt.Errorf("found more than one head for channel")
+	if len(headCandidates) > 1 {
+		var p string
+		for _, v := range headCandidates {
+			if v.sourceInfo.Package != p {
+				p = v.sourceInfo.Package
+			} else {
+				// if find multi heads in the same package, return error
```

Comment:
This would only fire if two _consecutive_ candidates were from the same package (`[A A B]` vs. `[A B A]`), which I don't think is the intention.

Reply:
@benluddy Thanks for your review and suggestions! I agreed. Update the error logs.